File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.cpp
class Solution {
public:
    int maxAbsoluteSum(vector<int>& nums) {
        int curSum = 0, minPre = 0, maxPre = 0, res = 0;
        for (int num: nums) {
            curSum += num;
            res = max(res, max(abs(curSum-maxPre), abs(curSum-minPre)));
            maxPre = max(maxPre, curSum);
            minPre = min(minPre, curSum);
        }
        return res;
    }
};

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        cur_sum = res = max_pre = min_pre = 0
        for num in nums:
            cur_sum += num
            res = max(abs(cur_sum-max_pre), abs(cur_sum-min_pre), res)
            max_pre = max(cur_sum, max_pre)
            min_pre = min(cur_sum, min_pre)
        return res